### PR TITLE
London class 8 - Hamra Khan - JS Core 2 - Week1

### PR DESCRIPTION
**Pull request by HamraKhan**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name: Hamra Khan
- Your City: London
- Your Slack Name: Hamra Khan

# Homework Details

- Module: JS Core 2
- Week: 1

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

File: 2-mandatory/2-currency-code-lookup.js [hunk 1]
```diff
@@ -18,7 +18,13 @@ const COUNTRY_CURRENCY_CODES = [
...
 
 function createLookup(countryCurrencyCodes) {
-  // write code here
+  let newCountryCurrencyCodes = Object.create(countryCurrencyCodes);
+  newCountryCurrencyCodes["GB"] = "GBP";
+  newCountryCurrencyCodes["DE"] = "EUR";
+  newCountryCurrencyCodes["NG"] = "NGN";
+  newCountryCurrencyCodes["MX"] = "MXN";
+
+  return newCountryCurrencyCodes;
```
Comment (Alirezabg) [inline 2-mandatory/2-currency-code-lookup.js:27]: Although the outcome is current, the function is not correct, you define the object manually.

File: 2-mandatory/1-recipes.js [hunk 2]
```diff
@@ -22,4 +22,12 @@
...
\ No newline at end of file
+const myFavoriteRecipe = {
+  title: 'Fajita wraps',
+  servings: 8,
+  ingredients: ['tortillas', 'chicken', 'mixed peppers', 'lettuce', 'cheese', 'potatoes']
+};
+
+console.log(myFavoriteRecipe.title);
+console.log(myFavoriteRecipe.servings);
+console.log(myFavoriteRecipe.ingredients);
```
Comment (Alirezabg) [inline 2-mandatory/1-recipes.js:33]: it needs five recipes and I think you should use a function to log it out.

File: 1-exercises/D-object-methods/exercise.js [hunk 3]
```diff
@@ -8,7 +8,9 @@
 */
 
 let student = {
-  // write code here
+  getName: function (name) {
+    console.log(`Student name: ${name}`);
+  }
```
Comment (Alirezabg) [inline 1-exercises/D-object-methods/exercise.js:13]: perfect!

File: 1-exercises/C-undefined-properties/exercise.js [hunk 4]
```diff
@@ -27,7 +32,8 @@ let user = {
 
 sayHelloToUser(user);
 
-// Example 3
+// // Example 3
+// 'return' is not being used inside the 'getName' function
```
Comment (Alirezabg) [inline 1-exercises/C-undefined-properties/exercise.js:36]: currect!

File: 2-mandatory/2-currency-code-lookup.js (hunk 1 above)
Comment (Alirezabg) [reply 2-mandatory/2-currency-code-lookup.js:27]: function createLookup(countryCurrencyCodes) {
  // write code here
  let currencyObject ={};
for (let currency of countryCurrencyCodes) {
  currencyObject[currency[0]] = currency[1];
}  
  return currencyObject;  
}

File: 2-mandatory/1-recipes.js (hunk 2 above)
Comment (ewintram) [reply 2-mandatory/1-recipes.js:33]: I agree, you can use an array method here to log the ingredients one by one, e.g. `forEach`

File: 2-mandatory/3-shopping-list.js [hunk 5]
```diff
@@ -10,16 +10,31 @@
...
+    name: "",
+    items: [],
+  };
+
+  //set name property of missingIngredients obj
+  missingIngredients.name = recipe.name;
+
+  recipe.ingredients.forEach(ingredient => {
+    if (!pantry.fridgeContents.includes(ingredient) && !pantry.cupboardContents.includes(ingredient)) {
+      missingIngredients.items.push(ingredient);
```
Comment (ewintram) [inline 2-mandatory/3-shopping-list.js:33]: great job!

**Review by ewintram — COMMENTED** [review]
great job! Ali beat me to some of the comments ;)

**Pull request closed without merging**